rewrite-java-test/src/test/java/org/openrewrite/java/RemoveUnusedImportsTest.java

```diff
@@ -1769,4 +1770,52 @@ public LocalDateTime toLocalDateTime(XMLGregorianCalendar cal) {
           )
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/4405")
+    @Test
+    void missingTypeReturnValueOfMethodWithArguments() {
+        rewriteRun(
+          spec -> spec.typeValidationOptions(TypeValidation.none()),
+          java(
+            """
+              import java.util.List;
+
+              public abstract class Foo1 {
+                  public abstract String m(String a);
+              }
+              """,
+            """
+              public abstract class Foo1 {
+                  public abstract String m(String a);
+              }
+              """
+          ),
+          java(
+            """
+              import java.util.GenClass1;
+
+              public abstract class Foo2 {
+                  public abstract GenClass1 m(String a);
+              }
+              """
```

Comment:
I'd assumed you don't want to see any change here; which is achieved by only having a single text block argument.